sql/core/src/test/scala/org/apache/spark/sql/SQLQueryTestHelper.scala

```diff
@@ -98,20 +98,24 @@ trait SQLQueryTestHelper extends Logging {
     }
   }
 
+  /**
+   * Uses the Spark logical plan to determine whether the plan is semantically sorted. This is
+   * important to make non-sorted queries test cases more deterministic.
+   */
+  protected def isSemanticallySorted(plan: LogicalPlan): Boolean = plan match {
```

Comment:
Is `isSemanticallySorted` only used in `getNormalizedQueryExecutionResult`?

Reply:
Yes. I moved it out because I want to use it in a subclass extending SQLQueryTestSuite, in this PR https://github.com/apache/spark/pull/44084.

Reply:
Got it.